### ISSUE TITLE
missing a step?

### ISSUE THREAD

**Issue by konsumer** [ericvicenti/csr-gen#2]
Am I missing something? In order to spin-up an HTTPS server, I would do something like this:

``` javascript
var privateKey = fs.readFileSync('privatekey.pem').toString();
var certificate = fs.readFileSync('certificate.pem').toString();
var server = https.createServer({key: privateKey, cert: certificate});
server.listen(8000);
```

To generate the key, csr, and cert I would run commands like this:

``` shell
openssl genrsa -out privatekey.pem 2048
openssl req -new -key privatekey.pem -out certrequest.csr -subj "/C=US/ST=California/L=San Fransisco/O=Loadtester/OU=Operations/CN=konsumer-mac.local/emailAddress=konsumer@jetboystudio.com"
openssl x509 -req -in certrequest.csr -signkey privatekey.pem -out certificate.pem
```

For a similar approach with your module:

``` javascript
csrgen(os.hostname(), {
    outputDir: __dirname,
    read: true,
    company: 'Loadtester',
    email: 'konsumer@jetboystudio.com'
}, function(err, keys){
    if (err) throw(err);    
    var server = https.createServer({key: keys.private, cert: NOCERT});
});
```

This lib does only the first 2 parts of the SSL signing process: generate a private key, and generate a csr.  Am I doing this wrong, or is it just missing the functionality to actually make a cert?  Would it be possible to use `crypto` to do all this, for a 100% native approach, so openssl command-line util is not needed?

**Comment by ericvicenti** [comment]
`csr-gen` does just that, generate the CSR. It is useful when buying SSL certificates for HTTPS websites etc. csr-gen does not generate self-signed certificates, which is what you are trying to do.

If you want to extend this module with an option for self-signing, I would be open to that. Although it module might need a new name, as it is currently just meant to generate the CSR.

It would be nice to use the `crypto` module, but I don't know how to use it to accomplish this.

**Comment by konsumer** [comment]
Yeh. I saw the [pem](https://github.com/andris9/pem), which does basically what your's does, and also does self-signing. I think I will use that for now.  It'd be neat to see something like this, all in native functions, especially for Windows deployments, but both modules appear to not do this. I will look into it.

For google, or your edification, here is how I did it with [pem](https://github.com/andris9/pem):

``` javascript
var pem = require('pem'),
    https = require('https');

pem.createCertificate({days:1, selfSigned:true}, function(err, keys){
  var server = https.createServer({keys.serviceKey, cert: keys.certificate});
  server.listen(8000);
});

```

This is the shortest/fastest/generate-all method, there are also methods for doing the individual parts, if you have part of the keys.
